File: backend/StreetCatCareCommunity/src/main/java/com/streetCat/pojo/Cat.java
package com.streetCat.pojo;

import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class Cat {

    private Long id;
    private String name;
    private String breed;
    private String gender;
    private Integer ageMonths;
    private String healthStatus;
    private String description;
    private String photos;
    private Boolean isNeutered;
    private String vaccinationStatus;
    private String status;
    private Long shelterId;
    private Long createdBy;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    private Long favorite;
    private Long comment;
    private Long view;
    private Long like;
}

File: frontend/StreetCatCareCommunity/src/types/cat.ts
interface cat {
  name: string;
  breed: string;
  gender: string;
  age_months: number;
  health_status: string;
  description: string;
  photos: string[];
  is_neutered: boolean;
  vaccination_status: string;
  status: string;
  shelter_id: string;
  created_by: string;
  id: string;
  created_at: string;
  favorite: number;
  comment: number;
  view: number;
  like: number;
  updated_at: string;
}

interface importCat {
    name: string;
    breed: string;
    gender: string;
    age_months: number;
    health_status: string;
    description: string;
    photos: string[];
    is_neutered: boolean;
    vaccination_status: string;
    status: string;
    shelter_id: string;
    created_by: string;
}

interface updateCat {
    name: string;
    breed: string;
    gender: string;
    age_months: number;
    health_status: string;
    description: string;
    photos: string[];
    is_neutered: boolean;
    vaccination_status: string;
    status: string;
    shelter_id: string;
    created_by: string;
    id: string;
}